### ISSUE TITLE
[HOLD for payment 2024-04-03] [$500] Scan - Inconsistency of flashlight/torch behavior in Scan tab

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#34457]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.24-4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
Prerequisite: Camera use permission is accepted previously
Android App
1. Login to android Expensify App
2. Tap FAB button
3. Tap Request money
4. Tap on flashlight icon to enable (verify nothing happens)
5. Tap on green circle to take a shoot with camera (verify photo is taken with flashlight is ON)
Mobile Web
1. Login to ND Expensify
2. Tap FAB button
3. Tap Request money
4. Tap on flashlight icon to enable (verify flashlight is ON)
5. Tap on flashlight icon to disable ((verify flashlight is OFF)

## Expected Result:
Flashlight should function exactly same behaviour both mobile apps and mobile web
Flashlight icon should be shown in Scan tab within all devices

## Actual Result:
Both Android and IOS apps works similarly; flashlight is only ON while taking a photo shoot
In Mobile Web, tapping on the flash icon turns the flashlight ON&OFF without camera taking a shoot

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/ea8878e3-55fd-4930-a067-3e5fffbdce3f

https://github.com/Expensify/App/assets/78819774/765741f9-fa33-404c-b4e5-57c5419116f1

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017684a08e378933c3</li>
        <li>Upwork Job ID: 1745866187913547776</li>
        <li>Last Price Increase: 2024-01-12</li>
<li>Automatic offers: </li>
<ul>
<li>s77rt | Reviewer | 28108708</li>
<li>tienifr | Contributor | 28108709</li>
</ul></li>
    </ul>
</details>

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The torch for native only appears when taking a picture, but for mWeb, the torch shows immediately when turned on.

### What is the root cause of that problem?
For native, the behavior is the flash is applied when taking the picture.
https://github.com/Expensify/App/blob/c8631be3c34cb7da835ff4db3ca1a8df7e19136f/src/pages/iou/request/step/IOURequestStepScan/index.native.js#L187-L190

But for web, we set up the torch like this:
https://github.com/Expensify/App/blob/c8631be3c34cb7da835ff4db3ca1a8df7e19136f/src/pages/iou/request/step/IOURequestStepScan/NavigationAwareCamera/index.js#L49-L57

### What changes do you think we should make in order to solve the problem?

For native, the camera can receive a torch prop that we can set to true.
```
torch={flash && device.hasTorch}
```

**Comment by s77rt** [comment]
@neonbhai Thanks for the proposal. I think we are looking to achieve the consistency here the other way around i.e. web should behave like native.

**Comment by s77rt** [comment]
@bernhardoj Thanks for the proposal. Same note as above ^

**Comment by s77rt** [comment]
@tienifr Thanks for the proposal. I agree on the RCA. Regarding the solution, can we improve this and use the track to construct [`ImageCapture`](https://developer.mozilla.org/en-US/docs/Web/API/ImageCapture). This Web API have a `takePhoto` method that supports [fillLightMode](https://developer.mozilla.org/en-US/docs/Web/API/ImageCapture/takePhoto#filllightmode) to choose if flash should be enabled.

PS: I think it would be better if we ship this feature upstream to react-webcam

**Comment by s77rt** [comment]
@neonbhai The mobile (native) behavior was implemented  first. Then support was added to web. I think it makes sense to match the web to the native. If you think the native design is incorrect please raise that to Slack (and tag me and the design team).

**Comment by s77rt** [comment]
Raised this on Slack https://expensify.slack.com/archives/C01GTK53T8Q/p1705368355349939 for more :eyes:

**Comment by s77rt** [comment]
I think we should proceed here with @tienifr's proposal.

🎀 👀 🎀 C+ reviewed
[Link to proposal](https://github.com/Expensify/App/issues/34457#issuecomment-1890362015)

**Comment by s77rt** [comment]
@tienifr still working on this

**Comment by strepanier03** [comment]
This is on hold for payment for another weekish so just clearing the overdue until then.

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: https://github.com/Expensify/App/pull/28618
- The offending PR has been commented on: https://github.com/Expensify/App/pull/28618#discussion_r1542148110
- A discussion in #expensify-bugs has been started: Not needed (unique bug)
- Determine if we should create a regression test for this bug: No, I don't think we need a test for this inconsistency

**Comment by strepanier03** [comment]
Looks like the automation didn't work here either. Handling payment stuff now.

**Comment by strepanier03** [comment]
All paid, closing!
